test/ruby/test_parse.rb

```diff
@@ -1052,6 +1052,22 @@ def test_yyerror_at_eol
     assert_syntax_error("    0b\n", /\^/)
   end
 
+  def test_unclosed_unicode_escape_at_eol_bug_19750
+    assert_separately([], "#{<<-"begin;"}\n#{<<~"end;"}")
```

Comment:
```suggestion
    assert_separately([], "#{<<-"begin;"}\n#{<<~'end;'}")
```

and you can remove doubled backslashes.